spec/controllers/api/v1/books_controller.rb

```diff
@@ -0,0 +1,46 @@
+require 'rails_helper'
+
+describe Api::V1::BooksController, type: :controller do
+  include_context 'Authenticated User'
+
+  describe 'GET #index' do
+    context 'When fetching all the books' do
+      let!(:books) { create_list(:book, 3) }
+
+      before do
+        get :index
+      end
+
+      it 'responses with the books json' do
+        expected = ActiveModel::Serializer::CollectionSerializer.new(
+          books, each_serializer: BookSerializer
+        ).to_json
+        expect(response_body.to_json) =~ JSON.parse(expected)
+      end
+
+      it 'responds with 200 status' do
+        expect(response).to have_http_status(:ok)
+      end
+    end
+  end
+
+  describe 'GET #show' do
+    context 'When fetching a book' do
+      let!(:book) { create(:book) }
+
+      before do
+        get :show, params: { id: book.id }
+      end
+
+      it 'responses with the book json' do
```

Comment:
Like we talked, can we add another spec to test when asking for a non existant book?
If you check the `api_controller`, you can find the rescue_from that we talked about...
In my opinion, I would change the response of the rescue from `:bad_request` to `:not_found`, and then make the spec.
If you have any doubt, ping me :)